Plugins/GitUIPluginInterfaces/GitConfigStatus.cs

```diff
@@ -0,0 +1,26 @@
+namespace GitUIPluginInterfaces
+{
+    /// <summary>
+    /// Status of running git config
+    /// https://git-scm.com/docs/git-config#_description This command will fail with non-
+    /// zero status upon error.Some exit codes are:
+    /// The section or key is invalid (ret= 1) this is returned when the key is not set also.
+    /// No section or name was provided (ret= 2).
+    /// The config file is invalid (ret=3).
+    /// The config file cannot be written (ret=4).
+    /// You try to unset an option which does not exist or you try to unset/set an
+    /// option for which multiple lines match(ret = 5).
+    /// You try to use an invalid regexp(ret = 6).
+    /// On success, the command returns the exit code 0.
+    /// </summary>
+    public enum GitConfigStatus
+    {
+        CannotWriteToConfigFile = 4,
```

Comment:
With this tight dependency on Git, the behaviour should be tested with real Git commands, where reasonable

Reply:
Read on further and note the tests.

Reply:
There are no tests to see that for instance `CannotWriteToConfigFile` is set when expected. This behavior can be changed in Git updates.

If this knowledge is not needed, no need to add the enums.

Reply:
I am providing all known git config values in the enum. Just because we use a custom config file work instead of actually reading from parsing git config list run doesn't mean the value can't be used in the future. The idea being we can call git config <key> value and get back an EffectiveGitSetting or what not. Still thinking about correct type name that makes sense for both potentials.   So I disagree. I am laying groundwork for both sets and gets. For now doing get but the full potential git config exit codes are represented in enum. I can look at git source code and look for what causes each exit code to better understand that. The rest I have makes sure that GetEffectiveGitSetting does not fail and also lists each config key with the exit code grouping. Running the code for every config so far shows exit code 1 and 0. If you want to pull my branch and run the test and verify...you can do so to provide verification of just 1 and 0.

Reply:
If you add something, it must be maintainable. If just 0/1 are expected, all others can be ignored, I see no direct need to handle them in the future. The extra values can be comments.

Reply:
I am providing the full enum at time of creation of the enum.  I have done the research of the git source code and understand each exit code and when it shows up.  I have documented it.  This is one of a few prs I am going to be doing in git config.  For example down the road I plan on still keeping file source classes from the git config file stuff but I want all git config reads to be read in by actually reading git config instead of us doing something custom with file reads and writes.  Still looking into the setting side where we want to set multiple keys.  That will be in a whole different pr/ issue.  I am laying the foundation for full git config driving and handling things we don't currently handle like multi valued configs and such.

Reply:
I think both points are valid. The risk of git changing these return codes is probably quite low, likely they'd break a _lot_ of people worldwide. At the same time, I find @gerhardol point about maintainability slightly more compelling, if we don't have an immediate use for some piece of code, I prefer not to add it. IMO it's totally valid to note all the possible return codes git may return for the config reading operation (thank you @vbjay for spending time finding those), but only expose the values we need. In the future, if a need to handle other return code arise, we can uncomment the necessary lines.